cmd/gateclient/client.go

```diff
@@ -152,15 +155,16 @@ func NewGateClient(flags *pflag.FlagSet) (*GatewayClient, error) {
 	}
 
 	// Api client initialization.
-	err = gateClient.Authenticate()
+	httpClient, err := gateClient.InitializeClient()
 	if err != nil {
-		util.UI.Error(fmt.Sprintf("OAuth2 Authentication failed."))
+		util.UI.Error(fmt.Sprintf("Could not initialize http client, failing."))
 		return nil, err
 	}
+	gateClient.HttpClient = httpClient
```

Comment:
Would it make sense to have `gateClient` set `HttpClient` on itself at the end of `InitializeClient` rather than have the caller do it?  (And possibly just skip returning it at all.)

Reply:
We use the client in two places - one when we do the raw `/login` call and once where we pass to the autogenerated Gate API library.